From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 19 Sep 2016 18:07:24 +0100
Subject: Revert "arm64: Define AT_VECTOR_SIZE_ARCH for ARCH_DLINFO"
Forwarded: not-needed

This reverts commit a7408eca94745c441acb1c6b55ebeb08b735bf1f which
was commit 3146bc64d12377a74dbda12b96ea32da3774ae07 upstream.
The bug doesn't cause a real problem yet and the fix changes
struct mm_struct which breaks the module ABI.

---
--- a/arch/arm64/include/asm/elf.h
+++ b/arch/arm64/include/asm/elf.h
@@ -140,7 +140,6 @@ typedef struct user_fpsimd_state elf_fpr
 
 #define SET_PERSONALITY(ex)		clear_thread_flag(TIF_32BIT);
 
-/* update AT_VECTOR_SIZE_ARCH if the number of NEW_AUX_ENT entries changes */
 #define ARCH_DLINFO							\
 do {									\
 	NEW_AUX_ENT(AT_SYSINFO_EHDR,					\
--- a/arch/arm64/include/uapi/asm/auxvec.h
+++ b/arch/arm64/include/uapi/asm/auxvec.h
@@ -19,6 +19,4 @@
 /* vDSO location */
 #define AT_SYSINFO_EHDR	33
 
-#define AT_VECTOR_SIZE_ARCH 1 /* entries in ARCH_DLINFO */
-
 #endif
